include/evmc/evmc.h

```diff
@@ -245,411 +281,469 @@ enum evmc_status_code
-                                  char const* name,
-                                  char const* value);
-
-
-/// EVM revision.
-///
-/// The revision of the EVM specification based on the Ethereum
-/// upgrade / hard fork codenames.
-enum evmc_revision {
+/**
+ * Configures the EVM instance.
+ *
+ *  Allows modifying options of the EVM instance.
+ *  Options:
+ *  - code cache behavior: on, off, read-only, ...
+ *  - optimizations,
+ *
+ *  @param evm    The EVM instance to be configured.
+ *  @param name   The option name. NULL-terminated string. Cannot be NULL.
+ *  @param value  The new option value. NULL-terminated string. Cannot be NULL.
+ *  @return       1 if the option set successfully, 0 otherwise.
+ */
+typedef int (*evmc_set_option_fn)(struct evmc_instance* evm, char const* name, char const* value);
+
+
+/**
+ * EVM revision.
+ *
+ *  The revision of the EVM specification based on the Ethereum
+ *  upgrade / hard fork codenames.
+ */
+enum evmc_revision
+{
     EVMC_FRONTIER = 0,
     EVMC_HOMESTEAD = 1,
     EVMC_TANGERINE_WHISTLE = 2,
     EVMC_SPURIOUS_DRAGON = 3,
     EVMC_BYZANTIUM = 4,
-    EVMC_CONSTANTINOPLE = 5,
+    EVMC_CONSTANTINOPLE = 5
 };
 
 
-/// Generates and executes machine code for given EVM bytecode.
-///
-/// All the fun is here. This function actually does something useful.
-///
-/// @param instance    A EVM instance.
-/// @param context     The pointer to the Host execution context to be passed
-///                    to callback functions. @see ::evmc_context.
-/// @param rev         Requested EVM specification revision.
-/// @param msg         Call parameters. @see ::evmc_message.
-/// @param code        Reference to the bytecode to be executed.
-/// @param code_size   The length of the bytecode.
-/// @return            All execution results.
+/**
+ * Generates and executes machine code for given EVM bytecode.
+ *
+ *  All the fun is here. This function actually does something useful.
+ *
+ *  @param instance    A EVM instance.
+ *  @param context     The pointer to the Host execution context to be passed
+ *                     to callback functions. @see ::evmc_context.
+ *  @param rev         Requested EVM specification revision.
+ *  @param msg         Call parameters. @see ::evmc_message.
+ *  @param code        Reference to the bytecode to be executed.
+ *  @param code_size   The length of the bytecode.
+ *  @return            All execution results.
+ */
 typedef struct evmc_result (*evmc_execute_fn)(struct evmc_instance* instance,
-                                             struct evmc_context* context,
-                                             enum evmc_revision rev,
-                                             const struct evmc_message* msg,
-                                             uint8_t const* code,
-                                             size_t code_size);
+                                              struct evmc_context* context,
```

Comment:
I kept this param packing. How do you like it? I think it's useful for documentation purposes.